### ISSUE TITLE
Save last iterate output

### ISSUE THREAD

**Issue by matthieutrs** [benchopt/benchopt#609]
Hello and thanks for the support! Computing and saving the metrics returned by an algorithm is great, but it may also be interesting to save the outputs of the algorithm (the path to saved output could be returned in the saved dataframe).

Example of use case: running several imaging algorithms with benchopt and saving the images returned by each algorithm.

**Comment by tomMoral** [comment]
Just talked to @sylvchev who has the same need for trained BCI models.
We could expose a function `save_last_run` that takes in an `output_dir` path and returns a `dict` with information that will be saved in the `DataFrame`.

**Comment by Badr-MOUFAD** [comment]
Thinking out load,

Inspired by the [newly added `warm_up` method](https://github.com/benchopt/benchopt/pull/602), 
one way to implement this feature is to add a new SolverBase method `post_run` that tacks the last result as input and performs some user-defined logic, in this case, it would be saving the output.

For instance, the following will be the user-defined method to save the output,

```python
class Solver(BaseSolver):
	...
	
	def post_run(self, result):
		path = "path_where_to_save_output"
		x = result["x"]
		
		with open(path, 'wb') as f:
			np.save(f, x)
	...
```

Any thoughts, @tomMoral @matthieutrs?

**Comment by matthieutrs** [comment]
Yes, I think it would be nice to have an easily customizable function, just like [hooks in torch lightning](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.core.hooks.ModelHooks.html). In my case, I'd be interested only in saving after the last run, but post_run would be fine too.

**Comment by tomMoral** [comment]
I would go for something like: `process_last_run_result`, but yes something customizable in the objective.

**Comment by paquiteau** [comment]
This is fixed by #722 !
